ogre/src/OgreLidarVisual.cc

```diff
@@ -171,77 +212,109 @@ void OgreLidarVisual::Update()
   // Process each point from received data
   // Every line segment, and every triangle is saved separately,
   // as a pointer to a DynamicLine
+  // This initializes and updates only the selected DynamicLine variables
   for (unsigned int j = 0; j < this->verticalCount; ++j)
   {
     horizontalAngle = this->minHorizontalAngle;
 
-    if (j+1 > this->dataPtr->rayLines.size())
+    if (this->dataPtr->lidarVisType == LidarVisualType::LVT_RAY_LINES ||
+        this->dataPtr->lidarVisType == LidarVisualType::LVT_TRIANGLE_STRIPS)
     {
-      // Ray Strips fill in between the line areas that intersect an object
-      std::shared_ptr<OgreDynamicLines> line =
-                  std::shared_ptr<OgreDynamicLines>(
-                              new OgreDynamicLines(MT_TRIANGLE_STRIP));
-      #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
-        line->setMaterial("Lidar/BlueStrips");
-      #else
-        Ogre::MaterialPtr mat =
+      if (j+1 > this->dataPtr->rayLines.size())
+      {
+        std::shared_ptr<OgreDynamicLines> line =
+                    std::shared_ptr<OgreDynamicLines>(
+                          new OgreDynamicLines(MT_LINE_LIST));
+
+        #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
+              line->setMaterial("Lidar/BlueRay");
+        #else
+            Ogre::MaterialPtr mat =
                 Ogre::MaterialManager::getSingleton().getByName(
-                                                "Lidar/BlueStrips");
-        line->setMaterial(mat);
-      #endif
-      std::shared_ptr<Ogre::MovableObject> mv =
+                                                "Lidar/BlueRay");
+            line->setMaterial(mat);
+        #endif
+        std::shared_ptr<Ogre::MovableObject> mv =
                 std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+        this->Node()->attachObject(mv.get());
+        this->dataPtr->rayLines.push_back(line);
+
+        if (this->dataPtr->lidarVisType == LidarVisualType::LVT_TRIANGLE_STRIPS)
+        {
+          line = std::shared_ptr<OgreDynamicLines>(
+                    new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+          #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
+            line->setMaterial("Lidar/LightBlueStrips");
+          #else
+            mat = Ogre::MaterialManager::getSingleton().getByName(
```

Comment:
[windows CI](https://build.osrfoundation.org/job/ign_rendering-pr-win/856/console) caught a build error here. This needs to be:

```
            Ogre::MaterialPtr mat = 
                Ogre::MaterialManager::getSingleton().getByName(
                "Lidar/BlueRay");

```